rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
+                                                                                           policyType,
+                                                                                           policyParameters,
+                                                                                           policyFileParameters,
+                                                                                           nodesRecoverable));
+    }
+
+    @Override
+    public String deployNodeSource(String sessionId, String nodeSourceName)
             throws RestServerException, ServiceException {
         return executeFunctionReturnStreamAsString(restClient -> restClient.deployNodeSource(sessionId,
                                                                                              nodeSourceName));
     }
 
     @Override
-    public String undeployNodeSource(final String sessionId, final String nodeSourceName, final boolean force)
+    public String undeployNodeSource(String sessionId, String nodeSourceName, boolean force)
             throws RestServerException, ServiceException {
         return executeFunctionReturnStreamAsString(restClient -> restClient.undeployNodeSource(sessionId,
                                                                                                nodeSourceName,
                                                                                                force));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * org.ow2.proactive_grid_cloud_portal.rm.client.RMService#getInfrastructures(java.lang.String)
-     */
-    public String getInfrastructures(final String sessionId) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.infrastructures(sessionId);
-            }
-        });
+    @Override
+    public String getInfrastructures(String sessionId) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.infrastructures(sessionId));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#getPolicies(java.lang.String)
-     */
-    public String getPolicies(final String sessionId) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.policies(sessionId);
-            }
-        });
+    @Override
+    public String getPolicies(String sessionId) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.policies(sessionId));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)